src/wazuh_db/wdb_agent.c

```diff
@@ -201,6 +210,8 @@ int wdb_remove_agent(int id) {
     sqlite3_bind_int(stmt, 1, id);
     result = wdb_step(stmt) == SQLITE_DONE;
     sqlite3_finalize(stmt);
+    wdb_close_global();
+
```

Comment:
**wdb_close_global()** may be called after `wdb_delete_agent_belongs(id);` because it also access the **global.db** database

Reply:
It shouldn't be necessary because `wdb_delete_agent_belongs` will open it again first.